### PR TITLE
feat: SLZ to encrypt cluster with KMS from another account

### PR DESCRIPTION
**Pull request by Ak-sky**
### Description

Added support to SLZ to encrypt cluster with KMS from external account

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by toddgiguere — CHANGES_REQUESTED** [review]
I think this is a good solution, using the CRN to get the three values needed.

You missed doing this exact same thing in the worker pool block in this same file, for the boot level encryption. I think it would be same solution.

File: cluster.tf [hunk 1]
```diff
@@ -108,8 +110,8 @@ resource "ibm_container_vpc_worker_pool" "pool" {
   worker_pool_name  = each.value.name
   flavor            = each.value.flavor
   worker_count      = each.value.workers_per_subnet
-  crk               = each.value.boot_volume_crk_name == null ? null : module.key_management.key_map[each.value.boot_volume_crk_name].key_id
-  kms_instance_id   = each.value.boot_volume_crk_name == null ? null : module.key_management.key_management_guid
+  crk               = each.value.boot_volume_crk_name == null ? null : regex("key:(.*)", module.key_management.key_map[each.value.boot_volume_crk_name].crn)[0]
+  kms_instance_id   = each.value.boot_volume_crk_name == null ? null : regex(".*:(.*):key:.*", module.key_management.key_map[each.value.boot_volume_crk_name].crn)[0]
```
Comment (ocofaigh) [inline cluster.tf:114]: your missing `kms_account_id` from here

Comment (Ak-sky) [reply cluster.tf:114]: 😕 Right. 
Resolved.

**Review by ocofaigh — APPROVED** [review]

**Comment by ocofaigh** [comment]
@Ak-sky The upgrade test failed with this:
```
2023-12-25T05:30:26.4879591Z         	Messages:   	Resource(s) identified to be updated 
2023-12-25T05:30:26.4880008Z         	            	Name: cluster
2023-12-25T05:30:26.4881184Z         	            	Address: module.roks_landing_zone.module.landing_zone.ibm_container_vpc_cluster.cluster["ocp-u-b5wtyk-workload-cluster"]
2023-12-25T05:30:26.4881918Z         	            	Actions: [update]
2023-12-25T05:30:26.4882515Z         	            	DIFF:
2023-12-25T05:30:26.4882841Z         	            	  Before: 
2023-12-25T05:30:26.4883532Z         	            		{"kms_config":"SECURE_VALUE_HIDDEN_HASH:-5370c9071f063d291e8b3ace5b35c663d7a0db918cfdca91104a47fc"}
2023-12-25T05:30:26.4884219Z         	            	  After: 
2023-12-25T05:30:26.4884872Z         	            		{"kms_config":"SECURE_VALUE_HIDDEN_HASH:-eeb6c449b7bbaf025911659722b06fd7bea6789cde75fb5333e88820"}
```

However I believe this is expected due to the nature of the change we are making to the kms_config. Since the value of `kms_config` is being masked, the only way to be sure the change is expected will be to manually simulate the upgrade test locally and check the changes identified in the plan for the upgrade. Is that something you can manually do?

BTW, we have an issue in our backlog to see if we can add support to our test framework to allow us to explicitially specify non secret values, such as `kms_config` in this case, but its not yet implemented.

**Comment by Ak-sky** [comment]
@ocofaigh for this should we skip the upgrade test?

**Review by ocofaigh — COMMENTED** [review]
Oh the upgrade is adding `account_id` to the KMS config. However that is not needed if the KMS is in the same account. We could probably avoid this update in place by passing `null` if the account is the same account.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**